packages/cmr-client/src/EarthdataLogin.ts

```diff
@@ -1,202 +1,155 @@
+      {
+        responseType: 'json',
+        username: username,
+        password: password,
+      });
+  } catch (error) {
+    if (error instanceof got.HTTPError) throw parseHttpError(error, 'create');
+    throw parseCaughtError(error);
   }
+  const response = PostTokenResponseBody.parse(rawResponse.body);
+  return response[0].access_token;
+};
 
-  /**
-   * The method for creating Earthdata Login token. This method sends a POST request
-   * to the Earthdata Login endpoint URL in order to create a token for the user. The users'
-   * username and password are sent as base64 encoded credentials as a header for
-   * authorization. If the users' credentials are accepted a token is created on their
-   * behalf and returned, if not, an error is thrown.
-   *
-   * @returns {Promise.<string>} the token
-   */
-  async createEDLToken(): Promise<string> {
-    const buff = Buffer.from(`${this.username + ':' + this.password}`).toString('base64');
-    const url = this.getEDLurl();
-    let response: EarthdataPostTokenResponse;
-    try {
-      response = await got.post(`${url}/api/users/token`,
-        {
-          headers: {
-            Authorization: `Basic ${buff}`,
-          },
-        }).json();
-    } catch (error) {
-      this.handleHttpError(error);
-    }
-    return response[0].access_token;
+/**
+  * This method is used for the cmrTokenSpec integration test in order to revoke the
+  * token that is created for testing.
+  *
+*/
+export const revokeEDLToken = async (username: string, password: string, edlEnv: string,
+  token: string): Promise<void> => {
+  try {
+    await got.post(`${getEdlUrl(edlEnv)}/api/users/revoke_token`,
+      {
+        searchParams: {
+          token,
+        },
+        username: username,
+        password: password,
+      });
+  } catch (error) {
+    if (error instanceof got.HTTPError) throw parseHttpError(error, 'revoke');
+    throw parseCaughtError(error);
   }
+};
 
-  /**
-   * This method is used for the cmrTokenSpec integration test in order to revoke the
-   * token that is created for testing.
-   *
-   */
-  async revokeEDLToken(
-    token: string
-  ): Promise<void> {
-    const buff = Buffer.from(`${this.username + ':' + this.password}`).toString('base64');
-    const url = this.getEDLurl();
-    try {
-      /* eslint-disable @typescript-eslint/no-unused-vars */
-      const response = await got.post(`${url}/api/users/revoke_token?token=${token}`,
-        {
-          headers: {
-            Authorization: `Basic ${buff}`,
-          },
-        }).json();
-      /* eslint-enable @typescript-eslint/no-unused-vars */
-    } catch (error) {
-      this.handleHttpError(error);
-    }
+/**
+ * The method that determines if a user has a token, to call the retrieveEDLToken function
+ * which makes an API call to the Earthdata Login endpoint, or to create the token with the
+ * createEDLToken function. Returns the token as a string.
+ *
+ * @returns {string} the token
+*/
```

Comment:
The indentation of these JSDocs needs to be fixed. The `*` here should be under the `*` above.